sphinx_gallery/scrapers.py

```diff
@@ -374,7 +383,7 @@ def save_figures(block, block_vars, gallery_conf):
     all_rst = ""
     prev_count = len(image_path_iterator)
     for scraper in _get_callables(gallery_conf, "image_scrapers"):
-        rst = scraper(block, block_vars, gallery_conf)
+        rst = scraper(block, block_vars, gallery_conf, file_conf)
```

Comment:
A question I have is how passing `file_conf` to custom scrapers should be handled? I'm guessing we cannot expect everyone using a custom scraper to add a new parameter. Is it enough to check if the scraper function has the corresponding param name?

Reply:
IIRC `block_vars` is a dict so can we just add it to `block_vars` like `file_conf["block_vars"]`? It's a bit of a misnomer to have file-level info in there but I think it's okay.

An alternative would be to use magic to inspect arguments but I don't think it's as clean

Reply:
Sure, can add this info to `block_vars`.

Just thinking ahead towards the other suggestion of having a block-specific setting, this info would still be in `block` at this point. Does it make sense to read this info. *before* it gets passed to the scraper and also store it in `block_vars`?